aiomusiccast/capabilities.py

```diff
@@ -0,0 +1,120 @@
+from abc import ABC
+from enum import Enum
+from typing import Any, Callable, Dict
+
+
+class EntityTypes(Enum):
+    """Type of the Capability."""
+    REGULAR = 1  # For major features
+    CONFIG = 2  # For features to configure the device or a zone
+    DIAGNOSTIC = 3  # For diagnostic values or settings
+    SYSTEM = 4  # Features for internal usage only
+
+
+class Capability(ABC):
+    """Base class for all capabilities."""
+    id: str
+    name: str
+    entity_type: EntityTypes
+    get_current: Callable
+
+    def __init__(self, capability_id, name, entity_type, get_current):
+        """
+        Initialize the base class and set general vars.
+        @param capability_id: Unique ID of this capability
+        @param name: Name that should be displayed in a UI
+        @param entity_type: Define of what type this capability is
+        @param get_current: Callable to get the current values of this capability
+        """
+        self.id = capability_id
+        self.name = name
+        self.entity_type = entity_type
+        self.get_current = get_current
+
+    @property
+    def current(self) -> Any:
+        return self.get_current()
+
+
+class SetableCapability(Capability, ABC):
```

Comment:
```suggestion
class SettableCapability(Capability, ABC):
```

https://www.dict.cc/?s=settable

Reply:
🙈